pkg/operator/ceph/nvmeof_recoverer/nvmeofstorage/controller.go

```diff
@@ -0,0 +1,116 @@
+	opcontroller "github.com/rook/rook/pkg/operator/ceph/controller"
+	"github.com/rook/rook/pkg/operator/ceph/reporting"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	controllerName = "nvmeofstorage-controller"
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", controllerName)
+
+var nvmeOfStorageKind = reflect.TypeOf(cephv1.NvmeOfStorage{}).Name()
+
+// Sets the type meta for the controller main object
+var controllerTypeMeta = metav1.TypeMeta{
+	Kind:       nvmeOfStorageKind,
+	APIVersion: fmt.Sprintf("%s/%s", cephv1.CustomResourceGroup, cephv1.Version),
+}
+
+var _ reconcile.Reconciler = &ReconcileNvmeOfStorage{}
+
+// ReconcileNvmeOfStorage reconciles a NvmeOfStorage object
+type ReconcileNvmeOfStorage struct {
+	client           client.Client
+	scheme           *runtime.Scheme
+	context          *clusterd.Context
+	opManagerContext context.Context
+	recorder         record.EventRecorder
+}
+
+// Add creates a new NvmeOfStorage Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, context *clusterd.Context, opManagerContext context.Context, opConfig opcontroller.OperatorConfig) error {
+	return add(mgr, newReconciler(mgr, context, opManagerContext))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, context *clusterd.Context, opManagerContext context.Context) reconcile.Reconciler {
+	return &ReconcileNvmeOfStorage{
+		client:           mgr.GetClient(),
+		context:          context,
+		scheme:           mgr.GetScheme(),
+		opManagerContext: opManagerContext,
+		recorder:         mgr.GetEventRecorderFor("rook-" + controllerName),
+	}
+}
+
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return errors.Wrapf(err, "failed to create %s controller", controllerName)
+	}
+	logger.Info("successfully started")
+
+	// Watch for changes on the NvmeOfStorage CRD object
+	cmKind := source.Kind(
+		mgr.GetCache(),
+		&cephv1.NvmeOfStorage{TypeMeta: controllerTypeMeta})
+	err = c.Watch(cmKind, &handler.EnqueueRequestForObject{}, opcontroller.WatchControllerPredicate())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcileNvmeOfStorage) Reconcile(context context.Context, request reconcile.Request) (reconcile.Result, error) {
+	logger.Debugf("reconciling NvmeOfStorage. Request.Namespace: %s, Request.Name: %s", request.Namespace, request.Name)
+
```

Comment:
remove the following lines

```
	if strings.Contains(request.Name, "nvmeofstorage") {
 		// TODO (cheolho.kang): Implement the reconclie logic later
 	}

 	// Return and do not requeue
 	logger.Debug("done reconciling")
```

Reply:
Okay,
Based on your comment, I think it's not relevant to the scope of this PR. I'll remove it.